cmd/k8s-operator/proxygroup.go

```diff
@@ -972,10 +971,9 @@ func (r *ProxyGroupReconciler) getDeviceInfo(ctx context.Context, staticEndpoint
 			device.TailnetIPs = ips
 		}
 
-		if hostname, _, ok := strings.Cut(string(m.stateSecret.Data[kubetypes.KeyDeviceFQDN]), "."); ok {
-			device.Hostname = hostname
-		}
-
+		// TODO(tomhjp): This is our input to the proxy, but we should instead
+		// read this back from the proxy's state in some way to more accurately
+		// reflect its status.
 		if ep, ok := staticEndpoints[device.Hostname]; ok && len(ep) > 0 {
```

Comment:
```suggestion
		if ep, ok := staticEndpoints[fmt.Sprintf("%s-nodeport", device.Hostname)]; ok && len(ep) > 0 {
```

Reply:
note - this may have been a small oversight on my part from the static endpoints PR